src/nodes/extension.rs

```diff
@@ -136,9 +135,11 @@ mod tests {
     fn rlp_extension_node_roundtrip() {
         let nibble = Nibbles::from_nibbles_unchecked(hex!("0604060f"));
         let val = hex!("76657262");
-        let extension = ExtensionNode::new(nibble, val.to_vec());
+        let mut child = vec![];
+        val.to_vec().as_slice().encode(&mut child);
+        let extension = ExtensionNode::new(nibble, child);
         let rlp = extension.as_ref().rlp(&mut vec![]);
-        assert_eq!(rlp, hex!("c88300646f76657262"));
+        assert_eq!(rlp, hex!("c98300646f8476657262"));
```

Comment:
Encoding looks to have been wrong, the extension node's child wasn't encoded as a string, but just plain hex without a header. Would have been decoded as `0x0604060f` + 4 single bytes, since each byte of `0x76657262` is <= `0x7F`